app/services/participants/change_schedule/validate_and_change_schedule.rb

```diff
@@ -26,6 +27,22 @@ def schedule
       def not_already_withdrawn
         errors.add(:participant_id, I18n.t(:withdrawn_participant)) if participant_profile_state&.withdrawn?
       end
+
+      def schedule_valid_with_pending_declarations
+        return unless user_profile
+
+        declarations = user_profile.participant_declarations.not_voided
+        declarations.each do |declaration|
+          milestone = schedule.milestone_for_declaration_type[declaration.declaration_type]
+          unless milestone.start_date.beginning_of_day < declaration.declaration_date
+            errors.add(:schedule_identifier, I18n.t(:schedule_invalidates_declaration))
+          end
+
+          unless declaration.declaration_date <= milestone.milestone_date.end_of_day
```

Comment:
```suggestion
          if declaration.declaration_date > milestone.milestone_date.end_of_day
```

What do `start_date` and `milestone_date` represent?

Reply:
`milestone_date` is the cut-off point for declarations. For the first milestone, it's October 31st. So all declarations for a milestone need to have a date before that. We wouldn't want the declaration date to be after the milestone date, because it would probably mean that the participant is not actually following the schedule they are given.

`start_date` is the other end of that - we don't want declarations to come in too early. For the first milestone it is some arbitraty date, usually 2 months before milestone date. For evety other milestone, it't the day after the end of previous milestone. At least in usual cases.

What it aims to model is "specific declarations should happen at specific time ranges, or the participant is not doing the course on the schedule we expect them to".